test/unit/lib/commands/drivers-util.test.ts

```diff
@@ -596,33 +592,59 @@ describe('drivers-util', () => {
-				expect.objectContaining({ primaryKeyName: 'driverId', sortKeyName: 'name' }),
-				expect.objectContaining({
-					preselectedId: 'preselected-driver-id',
-					promptMessage: 'Select a driver to install.',
-				}))
+		expect(selectFromListMock).toHaveBeenCalledTimes(1)
+		expect(selectFromListMock).toHaveBeenCalledWith(command,
+			expect.objectContaining({ primaryKeyName: 'driverId', sortKeyName: 'name' }),
+			expect.objectContaining({
+				preselectedId: 'preselected-driver-id',
+				promptMessage: 'Select a driver to install.',
+			}))
 
-			const drivers = [{ name: 'driver' }] as DriverChannelDetailsWithName[]
-			const listAssignedDriversWithNamesSpy = jest.spyOn(driversUtil, 'listAssignedDriversWithNames')
-				.mockResolvedValueOnce(drivers)
+		const drivers = [{ name: 'driver' }] as DriverChannelDetailsWithName[]
+		const listAssignedDriversWithNamesSpy = jest.spyOn(driversUtil, 'listAssignedDriversWithNames')
+			.mockResolvedValueOnce(drivers)
 
-			const listItems = selectFromListMock.mock.calls[0][2].listItems
+		const listItems = selectFromListMock.mock.calls[0][2].listItems
 
-			expect(await listItems()).toBe(drivers)
+		expect(await listItems()).toBe(drivers)
 
-			expect(listAssignedDriversWithNamesSpy).toHaveBeenCalledTimes(1)
-			expect(listAssignedDriversWithNamesSpy).toHaveBeenCalledWith(client, 'channel-id')
-		})
+		expect(listAssignedDriversWithNamesSpy).toHaveBeenCalledTimes(1)
+		expect(listAssignedDriversWithNamesSpy).toHaveBeenCalledWith(client, 'channel-id')
+	})
+
+	test('chooseInstalledDriver presents user with list of drivers with names', async () => {
+		const command = { client } as APICommand<typeof APICommand.flags>
+		selectFromListMock.mockResolvedValueOnce('chosen-driver-id')
+		stringTranslateToIdMock.mockResolvedValueOnce('preselected-driver-id')
+
+		expect(await driversUtil.chooseInstalledDriver(command, 'hub-id', 'prompt message', 'command-line-driver-id'))
+			.toBe('chosen-driver-id')
+
+		expect(stringTranslateToIdMock).toHaveBeenCalledTimes(1)
+		expect(stringTranslateToIdMock).toHaveBeenCalledWith(
+			expect.objectContaining({ primaryKeyName: 'driverId', sortKeyName: 'name' }),
+			'command-line-driver-id', expect.any(Function))
+		expect(selectFromListMock).toHaveBeenCalledTimes(1)
+		expect(selectFromListMock).toHaveBeenCalledWith(command,
+			expect.objectContaining({ primaryKeyName: 'driverId', sortKeyName: 'name' }),
+			expect.objectContaining({
+				preselectedId: 'preselected-driver-id',
+				promptMessage: 'prompt message',
+			}))
+		expect(apiHubdevicesListInstalledMock).toHaveBeenCalledTimes(0)
+
+		const listItems = stringTranslateToIdMock.mock.calls[0][2]
+		apiHubdevicesListInstalledMock.mockResolvedValueOnce([installedDriver])
+
+		expect(await listItems()).toStrictEqual([installedDriver])
+
+		expect(apiHubdevicesListInstalledMock).toHaveBeenCalledTimes(1)
+		expect(apiHubdevicesListInstalledMock).toHaveBeenCalledWith('hub-id')
```

Comment:
The last two tests in this file are quite large and could benefit from being split into multiple test cases.